blackbox/suites/customer/manus_perf/module.py

```
@@ -1349,7 +1349,7 @@ def run_extra_sample(
             checkout = checkout_parent / "repo"
             phases.append(self.run_phase(ctx, "clone", [["git", "clone", "--no-checkout", str(repo["url"]), str(checkout)]], checkout_parent, 1800, sample_env, shell=False, name=f"manus-extra-local-{repo['id']}-{profile}-run-{run_index}-clone"))
             phases.append(self.run_phase(ctx, "checkout", [["git", "-C", str(checkout), "checkout", "--detach", commit]], checkout_parent, 1800, sample_env, shell=False, name=f"manus-extra-local-{repo['id']}-{profile}-run-{run_index}-checkout"))
-            phases.append(self.run_phase(ctx, "install", [str(c) for c in repo.get("install", ["corepack enable", "corepack pnpm install --frozen-lockfile"])], checkout, 1800, sample_env, shell=True, name=f"manus-extra-local-{repo['id']}-{profile}-run-{run_index}-install"))
+            phases.append(self.run_phase(ctx, "install", [str(c) for c in repo.get("install", ["corepack prepare pnpm@10.33.0 --activate", "corepack pnpm install --frozen-lockfile"])], checkout, 1800, sample_env, shell=True, name=f"manus-extra-local-{repo['id']}-{profile}-run-{run_index}-install"))
```

Ruff (0.15.18)

[error] Function call with `shell=True` parameter identified, security issue

(S604)